Adding Scav/Extra Units to AI -- Fixes Early E/M Stalling due to Mod Options

Felnious opened the pull request:
Fixed Issues with Scavs Mod Options making AI Stall to Early Regarding Dropping a T3 EConverter stalling Energy and Metal before 10 Mins. Allowing players to use the Scav Option without having weak AI opponents at the Start of the Game (T2 Transition).

Changed Timing of |
*Arm/Cor/Legion - T3 EConv to Minimum 30min Timer.

*Arm/Cor/Legion - T3 Afus to Minimum 30min Timer.

Fixed and Placed within |
*Easy/Med/Hard/Aggressive

Script Changes |
Added Scav and Extra Unit Mod options to Init.AS for loading behaviour files Required.

Config Changes |
Added Extra Units and Scav Units behaviour Files.

Future Changes |
*Will possibly be adding these to Economy files at a later date, as well as build orders for future AI Economic Gains. For now just stopping Stalling of Energy and Metal before 30 Mins allowing for Games with Mod Options to not Feel so Weak and Lame!

*Will be adding Air Extra units at a Later date. Due to not being finished with Air Overhaul.

rlcevg commented:
Early commits reference jsons changes that are not present as summary change.
Is that intended?

Felnious commented:
The top 2 above merge shouldn't be included, those were already merged earlier and idk why those are showing up again in this one.

Felnious commented:
### Work done
Added Legsy to Legion Factory.json and Factory.as
Created New behaviours for Legion Sea Units and added them to All Difficulties Behaviour.json
Updated Economy_leg.json with new Water Economy
Updated Build_Chain.json with new Water Defenses for Legion
Fixed Armada / Cortex unused Unit Build_Chain.json

This also has the PR from https://github.com/beyond-all-reason/Beyond-All-Reason/pull/5864
